### ISSUE TITLE
Two array errors with `read`

### ISSUE THREAD

**Issue by thorstenkampe** [koalaman/shellcheck#1676]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2128, SC2178
- My shellcheck version (`shellcheck --version` or "online"):  0.7.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
```sh
#! /usr/bin/env bash

read -erp '- Host [localhost]:' host
read -erp '- Database:' db
read -eri "${ts-$db}" USER

host=${host:-localhost}
```

#### Here's what shellcheck currently says:
```
Line 5:
read -eri "${ts-$db}" USER
                ^-- SC2128: Expanding an array without an index only gives the first element.

Line 7:
host=${host:-localhost}
^-- SC2178: Variable was used as an array but is now assigned a string.
```
#### Here's what I wanted or expected to see:
The snippet reads three variables, none of which are of type array (`-a`). So there shouldn't be any error
